Applications/spire/source/book_view/quote_panel_indicator_widget.cpp

```diff
@@ -0,0 +1,48 @@
+#include "spire/book_view/quote_panel_indicator_widget.hpp"
+#include <QPainter>
+
+using namespace Spire;
+
+namespace {
+  constexpr auto FADE_TIME_MS = 400;
+
+  double map_to(double value, double a1, double a2, double b1, double b2) {
+    return ((value - a1) * ((b2 - b1) / (a2 - a1))) + b1;
+  }
+}
+
+QuotePanelIndicatorWidget::QuotePanelIndicatorWidget(QWidget* parent)
+    : QWidget(parent),
+      m_color(Qt::black) {
+  m_animation_timer.setInterval(33);
+  connect(&m_animation_timer, &QTimer::timeout, this,
+    &QuotePanelIndicatorWidget::on_animation_timer);
+}
+
+void QuotePanelIndicatorWidget::set_color(const QColor& color) {
+  m_color = color;
+  update();
+}
+
+void QuotePanelIndicatorWidget::animate_color(const QColor& color) {
+  m_color = color;
+  m_animation_start = std::chrono::steady_clock::now();
+  m_animation_timer.start();
```

Comment:
Do not wait 33ms before animating the color. Animate the color immediately.

Reply:
I missed a call to update() to force the widget to redraw. Was there something else? Otherwise, animate_color() is essentially frame 1 of the animation.

Reply:
```animate_color``` isn't performing any animation, it's waiting for a timer to expire.